### PR TITLE
Run ddp_spawn dataloader checks on windows

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes current failing windows tests

The fix is removing the `on_windows` skip. I don't know why we were skipping it.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Comment by Borda** [comment]
@carmocca does this affect Horovod somehow?

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by carmocca** [comment]
> @carmocca does this affect Horovod somehow?

No, the Horovod failure is unrelated. Probably another flaky test

**Review by kaushikb11 — APPROVED** [review]

**Review by Borda — CHANGES_REQUESTED** [review]
seems you fixed Win but broke Linux

File: tests/models/test_horovod.py [hunk 1]
```diff
@@ -240,6 +240,7 @@ def validation_step(self, batch, *args, **kwargs):
     tpipes.run_model_test_without_loggers(trainer_options, model)
 
 
+@pytest.mark.skip('TODO: flaky test - Fatal Python error: Aborted')
```
Comment (Borda) [inline tests/models/test_horovod.py:243]:
```suggestion
# todo: need to be fixed :]
@pytest.mark.skip('TODO: flaky test - Fatal Python error: Aborted')
```

File: tests/models/test_horovod.py [hunk 2]
```diff
@@ -272,7 +273,7 @@ def get_optimizer_params(optimizer):
     assert get_model_params(model.discriminator) == get_optimizer_params(trainer.optimizers[1])
 
 
-@pytest.mark.skipif(reason="CI agent.jobstatus=Succeeded: Permission denied")
+@pytest.mark.skip(reason="TODO: CI agent.jobstatus=Succeeded: Permission denied")
```
Comment (Borda) [inline tests/models/test_horovod.py:276]:
```suggestion
# todo: need to be fixed :]
@pytest.mark.skip(reason="TODO: CI agent.jobstatus=Succeeded: Permission denied")
```

File: tests/models/test_horovod.py [hunk 3]
```diff
@@ -322,7 +323,7 @@ def training_epoch_end(self, outputs) -> None:
     horovod.run(hvd_test_fn, np=2)
 
 
-@pytest.mark.skipif(reason="CI agent.jobstatus=Succeeded: Permission denied")
+@pytest.mark.skip(reason="TODO: CI agent.jobstatus=Succeeded: Permission denied")
```
Comment (Borda) [inline tests/models/test_horovod.py:326]:
```suggestion
# todo: need to be fixed :]
@pytest.mark.skip(reason="TODO: CI agent.jobstatus=Succeeded: Permission denied")
```

File: tests/models/test_horovod.py [hunk 4]
```diff
@@ -153,7 +153,7 @@ def test_horovod_multi_gpu_grad_by_value(tmpdir):
 
 # https://discuss.pytorch.org/t/torch-cuda-amp-vs-nvidia-apex/74994
```
Comment (Borda) [inline tests/models/test_horovod.py:154]:
```suggestion
# todo: need to be fixed :]
# https://discuss.pytorch.org/t/torch-cuda-amp-vs-nvidia-apex/74994
```

**Review by Borda — APPROVED** [review]
